app/server/static/components/annotator_qanda.vue

```diff
@@ -0,0 +1,179 @@
+<template lang="pug">
+  div(@click="setSelectedRange")
+    span.text-sequence(
+      v-for="r in chunks"
+      v-bind:class="getChunkClass(r)"
+      v-bind:style="getChunkStyle(r)"
+    ) {{ textPart(r) }}
+      button.delete.is-small(v-if="r.label==1", v-on:click="removeAnswer(r)")
+</template>
+
+<script>
+export default {
+  props: {
+    text: {
+      type: String,
+      default: '',
+    },
+    entityPositions: {
+      type: Array, // [{'startOffset': 10, 'endOffset': 15, 'label_id': 1}]
+      default: () => [],
+    },
+  },
+
+  data: () => ({
+    startOffset: 0,
+    endOffset: 0,
+  }),
+
+  computed: {
+    sortedEntityPositions() {
+      /* eslint-disable vue/no-side-effects-in-computed-properties */
+      this.entityPositions = this.entityPositions.sort((a, b) => a.start_offset - b.start_offset);
+      return this.entityPositions;
+      /* eslint-enable vue/no-side-effects-in-computed-properties */
+    },
+
+    chunks() {
+      const res = [];
+      let left = 0;
+      let e;
+      for (let i = 0; i < this.sortedEntityPositions.length; i++) {
```

Comment:
Can this be simplified with a for-of loop?

```js
for (let e of this.sortedEntityPositions) {
  // ...
}
```

Reply:
good idea. Just modified your way